File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k % nums.length; 
        
        ArrayList<Integer> newArray = new ArrayList<>();
        
        for (int i = nums.length - k; i < nums.length; i++) {
            newArray.add(nums[i]);
        }
        
        for (int i = 0; i < nums.length - k; i++) {
            newArray.add(nums[i]);
        }
        
        for (int i = 0; i < nums.length; i++) {
            nums[i] = newArray.get(i);
        }

        System.out.println(Arrays.toString(nums));
    }
}

File: 0189-rotate-array/0189-rotate-array.py
from collections import deque

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        new=deque(nums)
        new.rotate(k)
        nums_list=list(new)
        nums[:] = nums_list
        return nums